discord/ext/commands/view.py

```diff
@@ -108,21 +108,50 @@ def __repr__(self):
 
 # Parser
 
+# map from opening quotes to closing quotes
+quotes = {
+    '"': '"',
+    "'": "'",
+    "«": "»",
+    "‘": "’",
+    "‚": "‛",
+    "“": "”",
+    "„": "‟",
+    "‹": "›",
+    "⹂": "⹂",
+    "「": "」",
+    "『": "』",
+    "〝": "〞",
+    "﹁": "﹂",
+    "﹃": "﹄",
+    "＂": "＂",
+    "＇": "＇",
+    "｢": "｣",
+    "«": "»",
+    "‹": "›",
+    "《": "》",
+    "〈": "〉",
+}
+all_quotes = set(quotes.keys()) | set(quotes.values())
```

Comment:
Same concept here. Not actually sure why this is used though.

Reply:
all_quotes is used to speed up lookups for escaping both opening and closing quotes since otherwise it would have to iterate through quotes.values()